src/Attachment.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry;
+
+final class Attachment
+{
+    public const CONTENT_TYPE_DEFAULT = 'application/octet-stream';
+
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var string
+     */
+    private $contentType;
+
+    /**
+     * @var string
+     */
+    private $data;
+
+    public function __construct(string $data, string $filename, string $contentType = self::CONTENT_TYPE_DEFAULT)
+    {
+        $this->data = $data;
+        $this->filename = $filename;
+        $this->contentType = $contentType;
+    }
+
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    public function getContentType(): string
+    {
+        return $this->contentType;
+    }
+
+    public function getData(): string
+    {
+        return $this->data;
+    }
+
+    /**
+     * Helper method to create an Attachment from a file in the filesystem.
+     */
+    public static function fromFile(string $filename, string $contentType = self::CONTENT_TYPE_DEFAULT): self
+    {
+        $data = file_get_contents($filename);
```

Comment:
Since we are dealing with PHP here... we are reading the full file into memory without checking if it's within limits of something we can sent to Sentry (Sentry allows at most 20MB for a compressed request, and at most 100MB of uncompressed attachments per event). Although we might let the server bounce our request instead of adding client side validation.

I wonder if it's possible to use streams to move the files instead of needing them in memory twice (once here and once when the payload is constructed). At the very least we should try to keep the memory footprint as low as possible until we are constructing the evelope.